guides/source/active_record_encryption.md

```diff
@@ -352,10 +497,9 @@ NOTE: By default, `active_record.encryption` configures a `DerivedSecretKeyProvi
 
 #### EnvelopeEncryptionKeyProvider
 
-Implements a simple [envelope encryption](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping) strategy:
+The [`EnvelopeEncryptionKeyProvider`](https://api.rubyonrails.org/classes/ActiveRecord/Encryption/EnvelopeEncryptionKeyProvider.html) implements a simple [envelope encryption](https://cloud.google.com/kms/docs/envelope-encryption) strategy, where the data is encrypted with a key, which in turn is also encrypted.
```

Comment:
Maybe link to Wikipedia instead of a commercial party?
```suggestion
The [`EnvelopeEncryptionKeyProvider`](https://api.rubyonrails.org/classes/ActiveRecord/Encryption/EnvelopeEncryptionKeyProvider.html) implements a simple [envelope encryption](https://en.wikipedia.org/wiki/Hybrid_cryptosystem#Envelope_encryption) strategy, where the data is encrypted with a key, which in turn is also encrypted.
```